android-components/components/support/remotesettings/src/main/java/mozilla/components/support/remotesettings/RemoteSettingsClient.kt

```diff
@@ -93,7 +75,26 @@ class RemoteSettingsClient(
     }
 
     /**
-     * Reads all records for a collection  found in the local storage.
+     * Updates the local storage with [records] data only if records changed from last time.
+     */
+    suspend fun writeLastModified(records: RemoteSettingsResponse, lastModified: ULong): RemoteSettingsResult =
+        withContext(Dispatchers.IO) {
+            try {
+                if (records.lastModified != lastModified) {
+                    val jsonObject = JSONObject()
+                    jsonObject.put("records", records)
+                    AtomicFile(file).writeString { jsonObject.toString() }
+                    RemoteSettingsResult.Success(records)
+                } else {
+                    RemoteSettingsResult.NoChanges // Return a specific result for the case where no changes are made
+                }
+            } catch (e: IOException) {
+                RemoteSettingsResult.DiskFailure(e)
+            }
+        }
```

Comment:
Couple things here:

1. I think this could call the existing `write` function.
2. I was under the impression that we were looking to add functionality to this client to fetch and then write if the response was newer than what was stored. The way this function is reading, it still seems like it would be the responsibility of the consumer to handle that logic.

For 2, I think we have a few options:
1. Expose the `lastModified` property from the client and let the consumer handle writing updates to storage based on whether the response was newer.
2. Keep this function as is. 
3. Change this function to be a new extension function with a name like `fetchAndWriteIfResponseUpdated` that does a fetch _and_ and update if the response is newer than the previous `lastModified`.

Personally, I would opt for the 3rd option. I am just not sure we're adding a ton of value by adding another function where the only real difference is the `if (records.lastModified != lastModified)` condition. A consumer could probably handle that themselves.

Reply:
I added etchAndWriteIfResponseUpdated extension and writeIfLastModified functions as they both can be utilized individually.